crates/modelardb_server/src/storage/uncompressed_data_manager.rs

```diff
@@ -587,10 +660,68 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn test_will_finish_unused_uncompressed_data_buffer() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let (mut metadata_manager, mut data_manager, model_table_metadata) =
+            create_managers(temp_dir.path()).await;
+
+        // Insert using insert_data_points() to increment the batch counter.
+        let mut timestamp = TimestampBuilder::new();
+        timestamp.append_slice(&[100, 200, 300]);
+
+        let mut field = ValueBuilder::new();
+        field.append_slice(&[100.0, 200.0, 300.0]);
+
+        let mut tag = StringBuilder::new();
+        tag.append_value("A");
+        tag.append_value("A");
+        tag.append_value("A");
+
+        let record_batch = RecordBatch::try_new(
+            model_table_metadata.schema.clone(),
+            vec![
+                Arc::new(timestamp.finish()),
+                Arc::new(field.finish()),
+                Arc::new(tag.finish()),
+            ],
+        )
+        .unwrap();
+
+        data_manager
+            .insert_data_points(&mut metadata_manager, &model_table_metadata, &record_batch)
+            .await
+            .unwrap();
+
+        assert_eq!(data_manager.active_uncompressed_data_buffers.len(), 1);
+        assert_eq!(data_manager.finished_uncompressed_data_buffers.len(), 0);
+        assert_eq!(
+            data_manager
+                .active_uncompressed_data_buffers
+                .values()
+                .next()
+                .unwrap()
+                .len(),
+            3
+        );
+
+        // Insert using insert_data_points() to finish unused buffers.
+        let record_batch = RecordBatch::new_empty(model_table_metadata.schema.clone());
```

Comment:
Minor but maybe rename to `empty_record_batch` so it does not use the same name as the above record batch.